lib/practice_exercises.rb

```diff
@@ -1,13 +1,46 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  list.each_with_index do |value, index|
+    if index > 0 
+      if value == list[index - 1] 
+        list.slice!(index) 
+      end
+    end
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
+
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
-end
 
+  min_string = strings.min { |a,b| a.length <=> b.length}
+  hi = min_string.length - 1
+  low = 0
+  mid = (hi + low)/2
+  commons = []
+
+  strings.each do |string|
+    if min_string == string.slice(low..hi)
+      commons << string
+    end
+  end
+  return min_string if commons.length == strings.length
+
+  until hi <= low
```

Comment:
Since you never change low, you may not need it.  This made me think of binary search and it confused me for a bit.